test/tests/control.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { processInSpecProfile } from "../../src"
+import fs from 'fs'
+import path from 'path'
+
+describe('The control functionality', () => {
+    const cookstyle_profile = processInSpecProfile(fs.readFileSync('test/sample_data/inspec/json/cookstyle-controls-profile.json', 'utf-8'))
+
+    cookstyle_profile.controls.forEach(control => {
+        // Write the new control to the controls folder
+        fs.writeFileSync(path.join('test/sample_data/', 'controls-test-results', `${control.id}.rb`), control.toRuby())
+    })
+    
+    // This check is comparing what the function "toRuby" is outputting with a small sample profile that has been formatted 
```

Comment:
Please start the TSDoc process here and expand it out at least as a simple lift and shift and start the minimal GitHub Action to generate the TSDocs - <http://tsdocs.org>